packages/compiler-core/src/transforms/hoistStatic.ts

```diff
@@ -242,10 +251,16 @@ function hasDynamicKeyOrRef(node: ElementNode): boolean {
   return !!(findProp(node, 'key', true) || findProp(node, 'ref', true))
 }
 
-function hasCachedProps(node: PlainElementNode): boolean {
+function hasCachedProps(
+  context: TransformContext,
+  node: PlainElementNode
+): boolean {
   if (__BROWSER__) {
     return false
   }
+  if (!context.cacheHandlers) {
```

Comment:
This check should be performed in all cases because `cacheHandlers` is not the only mechanism that uses the cache, another example is `v-once`, and we may add more in the future.